### DISCUSSION TITLE
SOLVED - [Feature] Recursive backup

### DISCUSSION THREAD

**Discussion by patienttruth** [immich-app/immich#9489] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi, after searching around to figure out why my photos from a third party app weren't being backed up I found that it appears Immich is not backing up folders nested within a parent folder, only the images within that parent folder.

This app is used to annotate photos and has foldering capabilities built into it to organize projects. The trouble is that Immich is not scanning those folders and I have to manually add each project folder to Immich backup. Is it intentional to not backup recursively from mobile, or perhaps an error on my end?

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by NicholasFlamy** [comment]
So afaik Android doesn't care if the folders are subfolders, it adds them as an album. So your options are to either add the individual albums, or to add the "Recents" album which basically selects all photos (ignore the name, just deselect everything else and select Recents and all photos in all folders picked up by the Android Media manager/system/thing will be backed up. 

![Screenshot_20240516_102935](https://github.com/immich-app/immich/assets/30300649/c365185e-e7af-4ba0-b636-af2eba2dbd81)

**Reply by patienttruth** [reply]
Ah, I understand. This isn't Immich, but an Android limitation.

Thanks for the idea, but recent isn't ideal because it's grabbing images from messaging apps, etc.

I will work within this limitation. I'll probably have to just make a point of adding new projects to Immich.

Maybe it's not a wide use case,  but an idea I had was to make a setting in Immich to add all albums within a folder. But perhaps that's not even something that can be done due to how Android presents the folders/albums.

**Reply by NicholasFlamy** [reply]
I could be wrong but I'm pretty sure it's an Android limitation. Theoretically an Optional direct file access feature could be added but it's more complicated and I think Google doesn't like apps doing that.

Also, you could add recents but exclude most everything else.
![Screenshot_20240516_115958](https://github.com/immich-app/immich/assets/30300649/d20234b1-25b1-49fb-9ac6-4fb0923bbb31)

**Reply by patienttruth** [reply]
But regarding the exclusions, wouldn't anything aggregated into recent be backed up? When I checked recent I saw it pulled photos from apps such as messenger and whatsap which would be undesirable to back up.

**Reply by NicholasFlamy** [reply]
Only the recent album can be seen as "duplicates" of photos in other albums. So having arbitrary album a included and arbitrary album b excluded would be pointless because assets can't overlap and this would be equivalent to only arbitrary album a being included. So the whole point of exclusions is for similar to your use case where new stuff should be included but not certain things like app pictures.

Also, I tested this just now. Include recents and exclude anything other than your project albums.

**Reply by patienttruth** [reply]
I understand now, the Immich exclusion feature is smarter than the way that "recents" is aggregating the photos. Thanks for the additional explanation.

I made the changes to exclude anything that wasn't previously included, and I unchecked anything I had previously included (all of those folders from the third party app). Then I only included Recent. This appears to work exactly as I hoped. Thank you again for your help!

(SOLVED!)
